test/diag2dwd.jl

```diff
@@ -217,6 +218,317 @@ end
     @test nparts(advdiffdp, :Op2) == 2
 end
 
+@testset "Decapode Composition" begin
+
+  """
+      function unique_by!(acset, column_names::Vector{Symbol})
+
+  Given column names from the same table, remove duplicate rows.
+  
+  WARNING: This function does not check if other tables index into the one
+  given. Removal of rows is performed with prejudice.
+
+  See also: [`unique_by`](@ref).
+
+  # Examples
+  ```julia-repl
+  julia> g = @acset Graph begin
+    V = 2
+    E = 3
+    src = [1,1,2]
+    tgt = [1,1,2]
+  end
+  julia> unique_by!(g, :E, [:src, :tgt])
+  ``` 
+  """
+  function unique_by!(acset, table::Symbol, columns::Vector{Symbol})
+    rows = columns .|> (x -> acset[x]) |> (x -> zip(x...)) |> collect
+    rem_parts!(acset, table,
+      setdiff(parts(acset, table),
+        unique(i -> rows[i], eachindex(rows))))
+    return acset
+  end 
+
+  """
+      function unique_by(acset, column_names::Vector{Symbol})
+
+  Given column names from the same table, return a copy of the acset with
+  duplicate rows removed. Removal of rows is performed with prejudice.
+
+  WARNING: This function does not check if other tables index into the one
+  given. Removal of rows is performed with prejudice.
+
+  See also: [`unique_by!`](@ref).
+
+  # Examples
+  ```julia-repl
+  julia> g = @acset Graph begin
+    V = 2
+    E = 3
+    src = [1,1,2]
+    tgt = [1,1,2]
+  end
+  julia> unique_by(g, :E, [:src, :tgt])
+  ``` 
+  """
+  function unique_by(acset, table::Symbol, columns::Vector{Symbol})
+    acset_copy = copy(acset)
+    unique_by!(acset_copy, table, columns)
+  end 
+
+  """
+      function compose_decapodes(decapodes_vars::Vector{Tuple{NamedDecapode{
```

Comment:
docstrings should be of the form:
```
"""    funcname(args...)

description.
"""
```

this makes them work correctly in the docs.